lib/player.js

```diff
@@ -2406,6 +2421,33 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
       this.loadEventManager_.listen(
           mediaElement, 'timeupdate', onVideoProgress);
       this.onVideoProgress_();
+      if (this.manifest_.nextUrl) {
+        if (this.config_.streaming.minCompletionPreloadNextUrl > 0 &&
+            this.config_.streaming.minCompletionPreloadNextUrl < 100) {
+          const onTimeUpdate = async () => {
```

Comment:
note(no action required): ~can the existing `this.completionPercent_` (https://github.com/shaka-project/shaka-player/blob/main/lib/player.js#L699) be used here?~

I was thinking that using the existing value might improve things, but since we need to do some work it doesn't really save anything.

It *might* be worth moving this into `onVideoProgress_` but it isn't a clearly better option.

Reply:
But it is not the same, because onVideoProgress_ does not work the same when launching the events.